api/src/router.rs

```diff
@@ -1,41 +1,43 @@
+use crate::endpoints::{book, identity, root};
 use crate::middleware;
+
 use warp::{filters::BoxedFilter, Filter, Reply};
 
 pub fn router() -> BoxedFilter<(impl Reply,)> {
-    warp::path::end()
+    let root = warp::path::end()
+        // GET - /
         .and(warp::get())
-        .and_then(crate::endpoints::root::root)
-        .boxed()
-        .or(warp::path("identity").and(
-            warp::path("oauth")
-                .and(
+        .and_then(root::root);
+    let identity = warp::path("identity").and(
+        warp::path("oauth")
+            .and(
+                warp::path("client_identifier")
+                    .and(warp::path::end())
                     // GET - /identity/oauth/client_identifier
```

Comment:
This comments are a bit confuse me, because they are not on top of the path but right in the middle.